githubdelivery/src/main/java/io/dockstore/githubdelivery/GithubDeliveryS3Client.java

```diff
@@ -80,29 +93,79 @@ public static void main(String[] args) throws IOException {
+            pushPayload = MAPPER.readValue(body, PushPayload.class);
             return pushPayload;
         } catch (JsonSyntaxException e) {
             LOG.error("Could not read github event from key {}", key, e);
             System.exit(1);
         }
         return null;
     }
+    private InstallationRepositoriesPayload getGitHubInstallationRepositoriesPayloadByKey(String body, String key) throws IOException, NoSuchKeyException {
+        try {
+            InstallationRepositoriesPayload installationRepositoriesPayload;
+            installationRepositoriesPayload = MAPPER.readValue(body, InstallationRepositoriesPayload.class);
+            return installationRepositoriesPayload;
+        } catch (JsonSyntaxException e) {
+            LOG.error("Could not read github event from key {}", key, e);
+            System.exit(1);
+        }
+        return null;
+    }
+    private void submitGitHubDeliveryEventsByDate(GithubDeliveryConfig config, String date) {
+
+        ListObjectsV2Request listObjectsV2Request = ListObjectsV2Request
+                .builder()
+                .bucket(bucketName)
+                .prefix(date)
+                .build();
+        ListObjectsV2Iterable listObjectsV2Iterable = s3Client.listObjectsV2Paginator(listObjectsV2Request);
+        for (ListObjectsV2Response response : listObjectsV2Iterable) {
+            response.contents().forEach((S3Object event) -> {
+                submitGitHubDeliveryEventsByKey(config, event.key());
+            });
+        }
+    }
+    private void submitGitHubDeliveryEventsByKey(GithubDeliveryConfig config, String key) {
+        ApiClient apiClient = setupApiClient(config.getDockstoreConfig().serverUrl(), config.getDockstoreConfig().token());
+        WorkflowsApi workflowsApi = new WorkflowsApi(apiClient);
+        String deliveryid = key.split("/")[1]; //since key is in YYYY-MM-DD/deliveryid format
+        try {
+            String body = getObject(key);
+            if (body.contains("\"action\":\"added\"") || body.contains("\"action\":\"removed\"")) {
+                InstallationRepositoriesPayload payload = getGitHubInstallationRepositoriesPayloadByKey(body, key);
+                workflowsApi.handleGitHubInstallation(payload, deliveryid);
+            } else if (body.contains("\"deleted\":true")) {
+                PushPayload payload = getGitHubPushPayloadByKey(body, key);
+                workflowsApi.handleGitHubBranchDeletion(payload.getRepository().getFullName(), payload.getSender().getLogin(), payload.getRef(), deliveryid, payload.getInstallation().getId());
+            } else {
+                PushPayload payload = getGitHubPushPayloadByKey(body, key);
+                workflowsApi.handleGitHubRelease(payload, deliveryid);
+            }
+        } catch (IOException e) {
+            LOG.error("Could not submit github event from key {}", key, e);
```

Comment:
This is inside a loop, unsure whether we want to exit or continue onwards. 
In your experimentation, are there any downsides to "over" submitting? 
I assume the web-service will just ignore you if you submit something that has already been processed?

Reply:
> In your experimentation, are there any downsides to "over" submitting?

There might be consequences to oversubmitting.  For example, consider an "added installation" event A and subsequent "removed installation" event B.  Suppose we resubmit A, but not B.  After we're done, the webservice thinks that the app is installed on the repo, but it's not.

Reply:
Fair, although in this case if there had not been an IOException, that is the event the tool had been instructed to send. 
So you're being accidentally saved. I'd be more ok with this, if we aborted on `ApiException` but I think `IOException` is less clear.

Reply:
Changed so it aborts on `ApiException` and logs on `IOException`